src/main/java/org/dmfs/iterators/SerialIterator.java

```diff
@@ -43,7 +43,7 @@
     @SafeVarargs
     public SerialIterator(final Iterator<E>... iterators)
     {
-        mIterators = iterators.clone();
+        mIterators = iterators;
```

Comment:
In this case you also need to update `hasNext`, because this class actually modifies the array. It "releases" an iterator which has been completely iterated to the GC by setting its place in the array to `null`. But that's probably premature optimization.